Components/SingleArticle.jsx

```diff
@@ -1,30 +1,49 @@
 
 import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { getSingleArticle } from "../src/utils/api";
+import { getSingleArticle, voteOnArticle } from "../src/utils/api";
+import Comments from "./Comments";
 
 const SingleArticle = () => {
+    const [votes, setVotes] = useState(0);
+
   const { articleId } = useParams();
   const [article, setArticle] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
   const [isError, setIsError] = useState(false);
 
+  
   useEffect(() => {
     getSingleArticle(articleId)
       .then((articleData) => {
         setArticle(articleData);
+        setVotes(articleData.votes)
         setIsLoading(false);
       })
       .catch((err) => {
         setIsError(true);
         setIsLoading(false);
       });
-  }, []);
+  }, [articleId]);
+
+  const handleVote = (voteType) => {
+    setVotes((prevVotes) => (voteType === 'up' ? prevVotes + 1 : prevVotes - 1));
+
+    voteOnArticle(articleId, voteType)
+    .then((response) => {
+      console.log('Vote successful:', response);
+    })
+      .catch((error) => {
+        setVotes((prevVotes) => (voteType === 'up' ? prevVotes - 1 : prevVotes + 1));
+        console.error('Failed to vote:', error);
+      });
+  };
```

Comment:
Ideally, we want to inform a user about the failed vote, but not with a console.log - the average user is very unlikely to see this